src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -134,26 +169,90 @@ public String processEngineType(final Command requestingCommand, final String ch
+            case Command.M1_VOLUME_BETWEEN_3000_3500_CM ->
+                    userProgress.setM1Volume(M1EngineVolume.BETWEEN_3000_AND_3500);
+            case M1_VOLUME_MORE_3500_CM -> userProgress.setM1Volume(M1EngineVolume.MORE_3500);
+        }
+        userProgressStorage.save(userProgress);
+    }
+
+    public String processTruckUnitClass(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+
+        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
 
         switch (requestingCommand) {
-            case Command.VOLUME_LESS_1000_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.LESS_1000);
-            case Command.VOLUME_BETWEEN_1000_2000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_1000_AND_2000);
-            case Command.VOLUME_BETWEEN_2000_3000_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_2000_AND_3000);
-            case Command.VOLUME_BETWEEN_3000_3500_CM ->
-                    userProgress.setVolumeOfEngine(VolumeOfEngine.BETWEEN_3000_AND_3500);
-            case VOLUME_MORE_3500_CM -> userProgress.setVolumeOfEngine(VolumeOfEngine.MORE_3500);
+            case TRUCK_UNITS_6_CLASS -> userProgress.setTruckUnitType(TruckUnitClass.TRUCK_UNITS_6_CLASS);
+            case TRUCK_UNITS_EXCEPT_6_CLASS -> userProgress.setTruckUnitType(TruckUnitClass.TRUCK_UNITS_EXCEPT_6_CLASS);
         }
         userProgressStorage.save(userProgress);
-        message = messagesCreator.getSummaryAnswer(userProgress);
-        return message;
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processTruckUnitWeight(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+        if (!UserProgressValidator.validateCommand(Command.TRUCK_UNIT_WEIGHT, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+
+        switch (requestingCommand) {
+            case TRUCK_UNITS_12_20_TONS -> userProgress.setTruckUnitWeight(TruckUnitWeight.FROM_12_TILL_20_TONS);
+            case TRUCK_UNITS_20_50_TONS -> userProgress.setTruckUnitWeight(TruckUnitWeight.FROM_20_TILL_50_TONS);
+        }
+
+        userProgressStorage.save(userProgress);
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processTrailersO4Type(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+        if (!UserProgressValidator.validateCommand(Command.TRAILERS_O4_TYPE, userProgress)) {
```

Comment:
Is command really should be hardocoded?
```suggestion
        if (!UserProgressValidator.validateCommand(requestingCommand, userProgress)) {
```